apps/registration/src/components/ImageInputBox.tsx

```diff
@@ -0,0 +1,100 @@
+import { useRef, useState } from 'preact/compat';
+
+export const ImageInputBox = (prop:{
+    name: string[]
+}) => {
+    const handleUpload = () => {}
+    const {name} = prop
+
+    //file in this obj
+    const [files, setFiles] = useState(null)
```

Comment:
state and setState of state that will store file should be props which names obj and setObj. We will use provided props name "obj" to be the default value incase there is auto-save which will fetch the file value from API, and we will use props name "setObj" to set the value of obj.